### PR TITLE
Add banner to section dialog about verified teachers

### PR DESCRIPTION
**Pull request by molly-moen**
This PR adds a new banner to the section dialog if CSA is selected and the user is not a verified teacher. 

Screen Shot 2022-07-20 at 12 30 53 PM

## Links
- jira ticket: [JAVA-634](https://codedotorg.atlassian.net/browse/JAVA-634)

## Testing story
Tested locally.

## Follow-up work
Per discussion in [this thread](https://codedotorg.slack.com/archives/C01EF4GJ9GE/p1658334276595709) we should consider reusing this banner for CSP and CSD, which also have some verified teacher requirements. We can potentially use the verified resources setting on the course to determine whether or not to show this banner. For now this is hard-coded to CSA due to limited time to get this in before school year start.

### CODE REVIEW

File: apps/src/templates/teacherDashboard/EditSectionForm.jsx [hunk 1]
```diff
@@ -141,11 +145,18 @@ class EditSectionForm extends Component {
...
       assignedUnitTextToSpeechEnabled,
+      assignedUnitRequiresVerifiedInstructor,
       assignedUnitName,
       localeCode,
       isNewSection,
-      showLockSectionField // DCDO Flag - show/hide Lock Section field
+      showLockSectionField,
+      isVerifiedInstructor // DCDO Flag - show/hide Lock Section field
     } = this.props;
+    console.log(courseOfferings[section.courseOfferingId]);
```
Comment (dmcavoy) [inline apps/src/templates/teacherDashboard/EditSectionForm.jsx]: remove debug

Comment (molly-moen) [reply apps/src/templates/teacherDashboard/EditSectionForm.jsx]: 🤦‍♀️ thanks!

File: apps/src/templates/teacherDashboard/EditSectionForm.jsx [hunk 2]
```diff
@@ -141,12 +145,19 @@ class EditSectionForm extends Component {
...
       isNewSection,
-      showLockSectionField // DCDO Flag - show/hide Lock Section field
+      showLockSectionField,
+      isVerifiedInstructor // DCDO Flag - show/hide Lock Section field
     } = this.props;
 
+    const courseDisplayName =
+      section && section.courseOfferingId
+        ? courseOfferings[section.courseOfferingId].display_name
+        : '';
```
Comment (sanchitmalhotra126) [inline apps/src/templates/teacherDashboard/EditSectionForm.jsx]: Just curious, is it possible to view the edit section portion of the dashboard if there's no section?

**Review by sanchitmalhotra126 — APPROVED** [review]
LGTM!

File: apps/src/templates/teacherDashboard/EditSectionForm.jsx (hunk 2 above)
Comment (molly-moen) [reply apps/src/templates/teacherDashboard/EditSectionForm.jsx]: seems like no, we assume it exists elsewhere, I'll update the check to be less complicated

**Pull request merged**